[RNMobile] Update code comment for clarity

SiobhyB opened the pull request:
Addresses feedback received at https://github.com/wordpress-mobile/WordPress-Android/pull/16507#discussion_r870269435

In https://github.com/wordpress-mobile/WordPress-Android/pull/16507, a conditional that communicates whether xPosts are enabled on a site was updated to check whether a site was hosted on WordPress.com. The reason for this change is that the conditional's `mIsXPostsCapable` boolean returns `true` for .org sites by default. As the app only support xPosts for WordPress.com sites using a P2-based theme, an unresponsive `+` icon was being displayed in the editor's toolbar for .org sites.

The updated conditional, including comment, was as follows:

```
// If this.mIsXPostsCapable has not been set, default to allowing xPosts
boolean enableXPosts = mSite.isUsingWpComRestApi() && (mIsXPostsCapable == null || mIsXPostsCapable);
```

[This feedback](https://github.com/wordpress-mobile/WordPress-Android/pull/16507#discussion_r870269435) suggested that the comment could be updated for clarity, given the change. As such, this PR proposes the following change:  

```
// this.mIsXPostsCapable may return true for non-WP.com sites, but the app only supports xPosts for P2-based
// WP.com sites so, gate with `isUsingWpComRestApi()`
// If this.mIsXPostsCapable has not been set, default to allowing xPosts.
```
## Regression Notes
1. Potential unintended areas of impact

As this is a change to a code comment, no user-facing regressions are predicted. The only negative, unintentional impact would be developer confusion, if the comment isn't clear.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

N/A

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

SiobhyB committed:
Update comment to better clarify code's purpose

hypest commented on WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:
```diff
@@ -2305,7 +2305,9 @@ private GutenbergPropsBuilder getGutenbergPropsBuilder() {
 
-        // If this.mIsXPostsCapable has not been set, default to allowing xPosts
+        // this.mIsXPostsCapable may return true for non-WP.com sites, but the app only supports xPosts for P2-based
+        // WP.com sites.
```
```suggestion
        // WP.com sites so, gate with `isUsingWpComRestApi()`
```

Bit of a nit but, WDYT about the above suggestion instead of the `If the site is using the WP.com API`? I find the separate sentece for the API gate slightly confusing, and the "If ..." is a bit verbose since it can be seen in the code that it's a conditional.

SiobhyB replied:
Sounds like a good iteration to me! I've removed `If the site is using the WP.com API` in https://github.com/wordpress-mobile/WordPress-Android/pull/16547/commits/5f087bff30e00809b36033e36e15bf4b72912c78 and will go ahead to commit your suggestion.

SiobhyB committed:
Improve comment conciseness

"If the site is using the WP.com API..." was too verbose, especially considering that was evident from reading the conditional itself. It's therefore been removed with this comment. Instead, the previous sentence will be iterated on in a following commit to clarify why the 'isUsingWpComRestApi' is used.

SiobhyB committed:
Revert "Improve comment conciseness"

This reverts commit 30ecde5a177f36f68fc8f685ba85e12181de55aa.

SiobhyB committed:
Improve comment conciseness

"If the site is using the WP.com API..." was too verbose, especially considering that was evident from reading the conditional itself. It's therefore been removed with this comment. Instead, the previous sentence will be iterated on in a following commit to clarify why the 'isUsingWpComRestApi' is used.

SiobhyB committed:
Clarify usage of 'isUsingWpComRestApi'

Co-authored-by: Stefanos Togoulidis <stefanostogoulidis@gmail.com>

hypest reviewed (approved):
LGTM!
